archive/imo/imo1969_q1.lean

```diff
@@ -0,0 +1,70 @@
+/-
+Copyright (c) 2020 Kevin Lacker. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kevin Lacker
+-/
+
+import tactic.linarith
+import tactic.norm_cast
+import tactic.ring
+
+open int
+open nat
+
+/-!
+# IMO 1969 Q1
+
+Prove that there are infinitely many natural numbers $a$ with the following property:
+the number $z = n^4 + a$ is not prime for any natural number $n$.
+
+The key to the solution is that you can factor z into the product of two polynomials,
+if a = 4*m^4.
+-/
+
+lemma factorization (m n : ℤ) : ((m - n)^2 + m^2) * ((m + n)^2 + m^2) = n^4 + 4*m^4 := by ring
+
+/-!
+To show that the product is not prime, we need to show each of the factors is at least 2,
+which nlinarith can solve since they are expressed as a sum of squares.
+-/
+lemma left_factor_large (m n : ℤ) (h: 1 < m) : 1 < ((m - n)^2 + m^2) := by nlinarith
+lemma right_factor_large (m n : ℤ) (h: 1 < m) : 1 < ((m + n)^2 + m^2) := by nlinarith
+
+/-!
+The factorization is over the integers, but we need the nonprimality over the natural numbers.
+-/
+
+lemma int_large (a : ℤ) (h : 1 < a) : 1 < a.nat_abs :=
+by exact_mod_cast lt_of_lt_of_le h le_nat_abs
+
+lemma int_not_prime (a b : ℤ) (c : ℕ) (h1 : 1 < a) (h2 : 1 < b) (h3 : a*b = ↑c) : ¬ prime c :=
```

Comment:
```suggestion
lemma int_not_prime (a b : ℤ) (c : ℕ) (h₁ : 1 < a) (h₂ : 1 < b) (h₃ : a*b = ↑c) : ¬ prime c :=
```
Just a minor stylistic suggestion, not necessary to adopt. We love our unicode in mathlib. :-) You can type these just as `\1`, etc.